include/CL/cl2.hpp

```diff
@@ -6850,6 +6931,22 @@ inline vector<vector<unsigned char>> cl::Program::getInfo<CL_PROGRAM_BINARIES>(c
     return binariesVectors;
 }
 
+#if CL_HPP_TARGET_OPENCL_VERSION >= 220
+// Template specialization for clSetProgramSpecializationConstant
+template <>
+inline cl_int cl::Program::setSpecializationConstant(cl_uint index, const bool &value)
+{
+    cl_uchar ucValue = value ? CL_UCHAR_MAX : 0;
```

Comment:
This is a bit of an arbitrary choice. Have you considered making this configurable? (We do have many configuration options already though...)

Reply:
The OpenCL spec says for [clSetProgramSpecializationConstant](https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_API.html#clSetProgramSpecializationConstant):

> If a specialization constant is a boolean constant, spec_value should be a pointer to a cl_uchar value. A value of zero will set the specialization constant to false; any other value will set it to true.

I take this to mean that the actual value being set is fairly arbitrary, so long as it is nonzero.  I don't have a strong opinion which value to pick.

Reply:
Thinking about this a bit more, I'm wondering whether we shouldn't align on Vulkan and use a 32-bit type for the host/device interface. See https://renderdoc.org/vkspec_chunked/chap9.html#pipelines-specialization-constants.

Reply:
I believe that this implementation with a `uchar` is consistent with the current OpenCL API spec.  I'd be OK changing this to a 32-bit type instead, but I think we'd need to update the OpenCL API spec as well.

Reply:
I hadn't spotted the mention of the size/values in the API spec and your first reply. No reason to change anything then.